test/e2e_node/services/kubelet.go

```diff
@@ -239,6 +273,32 @@ func (e *E2EServices) startKubelet() (*server, error) {
 		cmdArgs = append(cmdArgs, "--hostname-override", framework.TestContext.NodeName)
 	}
 
+	// Write config file or flags, depending on whether --generate-kubelet-config-file was provided
+	if genKubeletConfigFile {
+		if err := writeKubeletConfigFile(kc, kubeletConfigPath); err != nil {
+			return nil, err
+		}
+		// add the flag to load config from a file
+		cmdArgs = append(cmdArgs, "--config", kubeletConfigPath)
+	} else {
+		// generate command line flags from the default config, since --generate-kubelet-config-file was not provided
+		fs := pflag.NewFlagSet("kubelet", pflag.ExitOnError)
+		options.AddKubeletConfigFlags(fs, kc)
+		// Walking the whole object results in a gigantic command line, so we just manually construct the flags we care about.
+		cmdArgs = append(cmdArgs, "--cgroup-root", fs.Lookup("cgroup-root").Value.String())
```

Comment:
Do this in a loop, and probably move into a helper function.

Reply:
will do